doc/bullet_collision_checker/src/bullet_collision_checker_tutorial.cpp

```diff
@@ -0,0 +1,312 @@
+  ROS_INFO("#####################################################");
+}
+
+void publishMarkers(visualization_msgs::MarkerArray& markers)
+{
+  // delete old markers
+  if (g_collision_points.markers.size())
+  {
+    for (int i = 0; i < g_collision_points.markers.size(); i++)
+      g_collision_points.markers[i].action = visualization_msgs::Marker::DELETE;
+
+    g_marker_array_publisher->publish(g_collision_points);
+  }
+
+  // move new markers into g_collision_points
+  std::swap(g_collision_points.markers, markers.markers);
+
+  // draw new markers (if there are any)
+  if (g_collision_points.markers.size())
+    g_marker_array_publisher->publish(g_collision_points);
+}
+
+void computeCollisionContactPoints(InteractiveRobot& robot)
+{
+  // move the world geometry in the collision world
+  Eigen::Isometry3d world_cube_pose;
+  double world_cube_size;
+  robot.getWorldGeometry(world_cube_pose, world_cube_size);
+  g_planning_scene->getWorldNonConst()->moveShapeInObject("world_cube", g_world_cube_shape, world_cube_pose);
+
+  collision_detection::CollisionRequest c_req;
+  collision_detection::CollisionResult c_res;
+  c_req.group_name = robot.getGroupName();
+  c_req.contacts = true;
+  c_req.max_contacts = 100;
+  c_req.max_contacts_per_pair = 5;
+  c_req.verbose = false;
+
+  g_planning_scene->checkCollision(c_req, c_res, *robot.robotState());
+
+  if (c_res.collision)
+  {
+    ROS_INFO("COLLIDING contact_point_count=%d", (int)c_res.contact_count);
+    if (c_res.contact_count > 0)
+    {
+      std_msgs::ColorRGBA color;
+      color.r = 1.0;
+      color.g = 0.0;
+      color.b = 1.0;
+      color.a = 0.5;
+      visualization_msgs::MarkerArray markers;
+
+      /* Get the contact ponts and display them as markers */
+      collision_detection::getCollisionMarkersFromContacts(markers, "panda_link0", c_res.contacts, color,
+                                                           ros::Duration(),  // remain until deleted
+                                                           0.01);            // radius
+      publishMarkers(markers);
+    }
+  }
+  else
+  {
+    ROS_INFO("Not colliding");
+
+    // delete the old collision point markers
+    visualization_msgs::MarkerArray empty_marker_array;
+    publishMarkers(empty_marker_array);
+  }
+}
+
+int main(int argc, char** argv)
+{
+  ros::init(argc, argv, "bullet_collision_tutorial");
+  ros::NodeHandle node_handle;
+
+  ros::AsyncSpinner spinner(1);
+  spinner.start();
+
+  moveit_visual_tools::MoveItVisualTools visual_tools("panda_link0");
+  visual_tools.setMarkerTopic("interactive_robot_markers");
+  ros::Publisher robot_state_publisher_2(node_handle.advertise<moveit_msgs::DisplayRobotState>("robot_state_before", 1));
+
+  {
+    // BEGIN_TUTORIAL
+    // The code starts with creating an interactive robot and a new planning scene.
+    InteractiveRobot interactive_robot;
+    g_planning_scene = new planning_scene::PlanningScene(interactive_robot.robotModel());
```

Comment:
Is there a reason to use raw pointers here?

Reply:
Either use smart pointer or give a reason why we are using raw pointers here

Reply:
I adapted the code from [the visualizing tutorial](https://github.com/ros-planning/moveit_tutorials/blob/master/doc/visualizing_collisions/src/visualizing_collisions_tutorial.cpp) for this tutorial and as it uses raw pointers I thought it would be fine.

Reply:
I replaced all raw pointers with smart ones.